mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -4631,6 +4633,51 @@ def remove_dataset_from_experiments(
 
             return dataset.to_mlflow_entity()
 
+    # ===================================================================================
+    # Helper Methods for Secrets & Endpoints
+    # ===================================================================================
+
+    def _get_entity_or_404(self, session, model_class, filters, entity_name):
```

Comment:
Should we call this `_get_entity_or_raise`? 404 is a http level contract.

Reply:
yep. "_or_raise" is definitely better. changed!